stunnel/stunnel.ml

```diff
@@ -29,6 +29,17 @@ let certificates_bundle_path = "/etc/stunnel/xapi-stunnel-ca-bundle.pem"
 let crl_path = "/etc/stunnel/crls"
 let verify_certificates_ctrl = "/var/xapi/verify_certificates"
 
+let get_verify_tls_certs () = Sys.file_exists verify_certificates_ctrl
+let set_verify_tls_certs = function
```

Comment:
@lindig I changed this so that the setting is backed by the file (this is actually better than using a ref, as you alluded to, because the setting persists between toolstack restarts)

Reply:
Could this create confusion? What is taking precedence? The file or the call? I think we need a clear way to enable and disable cert checking without changing code to enable testing.

Reply:
Well the file is the single source of truth, and this function is just a wrapper for creating/destroying the file (so you don't need to change the code to enable/disable checking)

Reply:
I'm still confused. If somebody else calls `set_verify` they need to have information already when to do this and so there is no need to persist this information in the file system. As a user, I could set the file but it could be overwritten by a call from `set`. Hence my question about preference. 

I don't want to make this complicated. If we have away to enable checking from xapi (using an xe command or configuration), that is all that is needed. An additional mechanism (like a file or environment variable) is only needed if we don't have such a mechanism but in that case preference needs to be spelled out.

Reply:
> If somebody else calls set_verify they need to have information already when to do this

Indeed, the only part of the toolstack that should be using this function in xapi

> there is no need to persist this information in the file system

I actually think it's a good idea to persist the state. If we don't persist the state, then I would be worried about the window between an 'xe-toolstack-restart', and xapi calling `set_verify_tls_certs`. If xapi happens to be is slow to start then some other part of the toolstack might use the stunnel library to open stunnels with the wrong verification setting

> As a user, I could set the file but it could be overwritten by a call from set.

I think this file should be the source of truth for xapi, and so xapi should not overwrite it, unless the user enables or disables verification via the API (NB on the REQ-403 branch the source of truth is the xapi local database, but we can change that when porting to master). Does that sounds reasonable to you?

Reply:
Wouldn't xapi's database be the single source of truth? as long as the database field I don't see the need to use a file. The database is persisted in any case and using a special file only for this usecase makes the system harder to understand. Am I missing something?

Reply:
If we are happy to accept this window that I mentioned above between a toolstack restart beginning, and xapi setting the verification flag in stunnel.ml then I agree that the file is an unnecessary layer